awx/main/migrations/0021_v330_create_new_rbac_roles.py

```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations
+from awx.main.migrations import ActivityStreamDisabledMigration
+from awx.main.migrations import _rbac as rbac
+from awx.main.migrations import _migration_utils as migration_utils
+
+
+class Migration(ActivityStreamDisabledMigration):
+
+    dependencies = [
+        ('main', '0020_v330_declare_new_rbac_roles'),
+    ]
+
+    operations = [
+        migrations.RunPython(migration_utils.set_current_apps_for_migrations),
+        migrations.RunPython(rbac.create_roles),
```

Comment:
does this create_roles script also establish the new parentage for everything? You're creating a new organization execute_role, so does all of the related JT execute_roles now become children after the migration, even though they were created before the migration?

Reply:
Yes, this method uses the `batch_role_ancestor_rebuilding` context decorator.